app/services/search/postgres/tag.rb

```diff
@@ -0,0 +1,13 @@
+module Search
+  module Postgres
+    class Tag
+      def self.search_documents(term)
```

Comment:
Here we are using the scope from above. Note that we can use the normal AR method chaining, so we can easily return attributes that weren't part of the search. I tucked `as_json` on the end to get a result similar to what the current search controller returns:

```ruby
# current ES code
[
  [0] {
      "hotness_score" => 175,
                 "id" => 7,
    "last_indexed_at" => "2020-12-23T05:22:45.878Z",
               "name" => "javascript",
         "rules_html" => nil,
      "short_summary" => nil,
          "supported" => true
  },
  [1] {
      "hotness_score" => 160,
                 "id" => 6,
    "last_indexed_at" => "2020-12-23T05:23:07.046Z",
               "name" => "java",
         "rules_html" => nil,
      "short_summary" => nil,
          "supported" => true
  }
]

# new PG code
[
  [0] {
    "hotness_score" => 175,
               "id" => 7,
             "name" => "javascript",
       "rules_html" => nil,
    "short_summary" => nil,
        "supported" => true
  },
  [1] {
    "hotness_score" => 160,
               "id" => 6,
             "name" => "java",
       "rules_html" => nil,
    "short_summary" => nil,
        "supported" => true
  }
]
```

The PG code is only missing the `last_indexed_at` attribute, but I suppose we could do something to get that back too. We also didn't need to pass in an explicit wildcard character since I enabled the prefix search option on the scope. 

/cc @rhymes

Reply:
cool! From a brief look it seems like we're using `last_indexed_at` only as stored metadata, added during indexing. We're not using it anywhere during query time. 

Probably solvable with https://github.com/Casecommons/pg_search#more-options, section "Specify additional attributes to be saved on the pg_search_documents table" but it requires new fields and the usage of `multisearchable`, not sure it's worth it for now

Reply:
Ah, I should have been more specific, I tagged you mostly because you asked about using multisearch and search scopes together. 😃 In this specific case, `last_indexed_at` wouldn't make sense, we don't have a dedicated `tsvector` column on this table (yet) so nothing is really "indexed".  The generated query builds the vector on the fly and uses the [`@@`](https://www.postgresql.org/docs/13/functions-textsearch.html) operator to check if it matches the query. An example query looks like this:

```sql
SELECT "tags".* FROM "tags" 
INNER JOIN 
  (SELECT "tags"."id" AS pg_search_id, tags.hotness_score AS rank 
   FROM "tags" 
   WHERE ((to_tsvector('simple', coalesce("tags"."name"::text, ''))) @@ (to_tsquery('simple', ''' ' || 'java' || ' ''' || ':*')))) AS pg_search_978c2f8941354cf552831b 
ON "tags"."id" = pg_search_978c2f8941354cf552831b.pg_search_id 
ORDER BY pg_search_978c2f8941354cf552831b.rank DESC, "tags"."id" ASC
```

I'm actually starting to doubt if this specific model needs FTS at all, the search scope could probably be replaced with a left-anchored `LIKE `query and an appropriate index.

Reply:
Probably the best way to know it is to generate lots of tags and to `EXPLAIN ANALYZE` the two queries. We have 73000 tags in production

Reply:
I mean we could do that, but when you look at the type of search that we perform here (prefix search of a single word against a list of words) it doesn't look like we need any advanced FTS features. But tbh this is probably outside the scope of this spike, optimizing single queries for one search type over the other can definitely wait. 😃